### PR TITLE
Manual DB Seed

### PR DESCRIPTION
**Pull request by maxdemaio**
This PR brings in a small `drawio` file to show how the tables are setup and a DB seed file. For the DB seed file it works with only 1 file (hardcoded filename) at a time and needs some cleanup but works nonetheless!

Below is an image of the successfully loaded `eng-deu.xml` dict into the database. This is a snippet of the `entries` table limited to 50 rows:
Screen Shot 2022-09-03 at 2 02 57 PM

Below is an image confirming our 500,000+ definitions got in safe and sound:
Screen Shot 2022-09-03 at 2 17 57 PM

### CODE REVIEW

**Comment by Nickersoft** [comment]
This is great! So what ended up being the bottleneck here? Is it the fact `Promise.all()` dispatches too many threads and it crashes the app? So they have to be run in sequence using for loops? What would be nice is if we could dump out the raw SQL statements that we could use to seed a production database without it taking as long haha.

File: prisma/manualSeed.ts [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+		// if exists, stop/return
+		if (dict != null) {
+			return;
+		}
+	}
+
+	// if source/target language don't exist, add them to the db
+	if (dbSourceLang == null) {
+		await prismaClient.$queryRaw`INSERT INTO "languages" ("code", "flag") VALUES (${sourceCode}, '"testFlag"')`;
```
Comment (Nickersoft) [inline prisma/manualSeed.ts:97]: Not sure why this specifically has to be a raw query. Couldn't you just use `prismaClient.language.create()`?

**Review by Nickersoft — CHANGES_REQUESTED** [review]

**Comment by maxdemaio** [comment]
> This is great! So what ended up being the bottleneck here? Is it the fact `Promise.all()` dispatches too many threads and it crashes the app? So they have to be run in sequence using for loops? What would be nice is if we could dump out the raw SQL statements that we could use to seed a production database without it taking as long haha. I think this is a good start, though – I would extract the logic to a shared function though, so that way we can have a single seed file that reads a bunch of XML dictionaries and then passes them to a seeder function.

I think you're right on the money with that one with why `Promise.all()` fails. The for-loops seem to do the trick too.

Luckily with:
```
const prismaClient = new prisma.PrismaClient({
	log: ['query', 'info', 'warn', 'error']
});
```

I think this prints out all of the raw SQL statements that Prisma is performing. I'll get this file cleaned/fixed up with your suggestions and then we could optimize from there 💪🏻

**Comment by Nickersoft** [comment]
Are you talking about a UML-style database diagram or something? I know Prisma has a `studio` command, but I don't think it does any diagramming: https://www.prisma.io/studio.

I'm also aware about the logging, though, it's a pain to try and command and paste all that haha. I wish Prisma supported like an SQL dump command. On the bright side, if we can get everything seeded locally we can just use `pgdump` to get all the data from the database as SQL.

**Comment by maxdemaio** [comment]
Yes I think so, just a small crows foot notation thing to show people how the DB is set up. Actually I didn't know about Prisma's `studio` command so I'll have to take a peek.

I made the changes you requested and made the `main()` function loop over all the files in our `dictionaries` directory 💪🏻 I pulled in some test ODXML dicts from [your examples](). Looks like the function now works like a charm:

Screen Shot 2022-09-05 at 8 04 03 AM

**Review by Nickersoft — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,38 +1,23 @@
...
 
-## Building
-
-Before creating a production version of your app, install an [adapter](https://kit.svelte.dev/docs#adapters) for your target environment. Then:
+Run the project:
 
-```bash
-npm run build
 ```
+yarn dev
```
Comment (Nickersoft) [inline README.md:18]: Maybe outside of the scope of this PR (overall LG! we should merge), but I think we should set up a docker-compose file that can spin up a Postgres image, then run our seeder to dump our base dictionaries into it to make dev a bit more seamless.

Also, after this merges, I'm curious how you'd like to coordinate building out the FE. I have some half-assed designs I did in Figma a long time ago but I think they need some revisiting 😅

Comment (maxdemaio) [reply README.md:18]: I'll merge this in now! 

The docker-compose file sounds like a great idea. Then developers could easily jump right into the code base! Lol for the frontend I'd be down to just share a Figma (Adobe now 😢) sketch file and do some wireframes of some sort.

**Pull request merged**